uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/manager-leave/components/manager-leave-calendar.js

```diff
@@ -10,11 +13,493 @@ define([
+        return vm.filteredContacts.filter(function (contact) {
+          return leaveRequests.find(function (request) {
+            return request.contact_id == contact.id;
+          });
+        });
+      }
+
+      return vm.filteredContacts;
+    };
+
+    /**
+     * Returns day name of the sent date(Monday, Tuesday etc.)
+     *
+     * @param  {string} date
+     * @return {string}
+     */
+    vm.getDayName = function (date) {
+      var day = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
+      return day[getDateObjectWithFormat(date).day()];
+    };
+
+    /**
+     * Returns the calendar information for a specific month
+     *
+     * @param  {int/string} contactID
+     * @param  {int} month
+     * @return {array}
+     */
+    vm.getMonthData = function (contactID, month) {
+      var contact = _.find(vm.managedContacts, function (contact) {
+        return contact.id == contactID
+      });
+
+      if (contact && contact.calendarData) {
+        var calendarData = contact.calendarData;
+        var datesForTheMonth = [],
+          dates = Object.keys(calendarData.days);
+
+        dates.forEach(function (date) {
+          if (moment(parseInt(date)).month() === month) {
+            datesForTheMonth.push(calendarData.days[date]);
+          }
+        });
+
+        return datesForTheMonth;
+      }
+    };
+
+    /**
+     * Decides whether sent date is a public holiday
+     *
+     * @param  {string} date
+     * @return {boolean}
+     */
+    vm.isPublicHoliday = function (date) {
+      return !!publicHolidays[getDateObjectWithFormat(date).valueOf()];
+    };
+
+    /**
+     * Labels the given period according to whether it's current or not
+     *
+     * @param  {AbsencePeriodInstance} period
+     * @return {string}
+     */
+    vm.labelPeriod = function (period) {
+      return period.current ? 'Current Period (' + period.title + ')' : period.title;
+    };
+
+    /**
+     * Refresh contacts and calendar data
+     */
+    vm.refresh = function () {
+      vm.loading.calendar = true;
+      loadContacts()
+        .then(function () {
+          loadLeaveRequestsAndCalender()
+            .then(function () {
+              vm.loading.calendar = false;
+            });
+        });
+    };
+
+    (function init() {
+      vm.loading.page = true;
+      //Select current month as default
+      vm.selectedMonths = [vm.months[moment().month()]];
+      $q.all([
+        loadAbsencePeriods(),
+        loadAbsenceTypes(),
+        loadPublicHolidays(),
+        loadRegions(),
+        loadDepartments(),
+        loadLocations(),
+        loadLevelTypes(),
+        loadStatuses(),
+        loadDayTypes()
+      ])
+        .then(function () {
+          return loadManagees();
+        })
+        .then(function () {
+          vm.legendCollapsed = false;
+          return loadLeaveRequestsAndCalender();
+        })
+        .then(function () {
+          vm.loading.page = false;
```

Comment:
can we keep this in finally because if there is an error then `vm.loading.page` will never be set to false?